internal/cmd/skupper/listener/nonkube/listener_delete_test.go

```diff
@@ -147,17 +145,40 @@ func TestCmdListenerDelete_Run(t *testing.T) {
 
 		t.Run(test.name, func(t *testing.T) {
 
+			createListenerResource(path, t)
 			cmd.listenerName = test.deleteName
 			cmd.namespace = test.namespace
 			cmd.listenerHandler = fs.NewListenerHandler(cmd.namespace)
 			cmd.InputToOptions()
 
 			err := cmd.Run()
 			if err != nil {
-				assert.Check(t, test.errorMessage == err.Error())
+				assert.Check(t, strings.HasSuffix(err.Error(), test.errorMessage))
 			} else {
 				assert.Check(t, err == nil)
 			}
 		})
 	}
 }
+
+func createListenerResource(path string, t *testing.T) {
+	listenerResource := v2alpha1.Listener{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "skupper.io/v2alpha1",
+			Kind:       "Listener",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-listener",
+			Namespace: "test",
+		},
+	}
+
+	listenerHandler := fs.NewListenerHandler("test")
+
+	defer listenerHandler.Delete("my-connector")
+
+	contentConnector, err := listenerHandler.EncodeToYaml(listenerResource)
+	assert.Check(t, err == nil)
+	err = listenerHandler.WriteFile(path, "my-connector.yaml", contentConnector, common.Sites)
```

Comment:
Two issues here: `my-connector.yaml` -> `my-listener.yaml` and `common.Sites` -> `common.Listeners`.